### PR TITLE
std.format: Add overview of format indicator

### PR DESCRIPTION
**Pull request by berni44**
Formatting decimals with `%b`, `%o` and `%x` is often referred to as writing this number in binary, octal and hexadecimal. But that is not, what (C-)users expect and what the current implementation produces: It is just an image of the underlying bits. That means `format!"%x"(-10);` results in `fffffff6` instead of `-a`. I tried to follow this with the descriptions.

A similar problem occurs with `%u`, which is often called as formatting as unsigned integer. This was needed with C, because C has no traits and cannot know if the argument was meant to be a signed or an unsigned integer, as far as I know. In D this does not make any sense and in the current implementation `%u` just produces the same results as `%d`. IMHO `%u` could be discarded completely. Yet, people think of it as a feature (e.g. #7913). To make the best out of it, I choose to describe it similar to `%b`, `%o` and `%x` as a (yet buggy implemented) representation of the bit pattern. This seems to me closest to the intention behind `%u`. An alternative would be to kick it into the long grass. Would be fine for me, but might be perceived as a breaking change.

### CODE REVIEW

File: std/format/package.d
Comment (Geod24) [inline std/format/package.d:151]:
```suggestion
expression surrounded by $(B %\() and $(B %\)). It is the minimum
requirement to format a value.
```
Perhaps ?

File: std/format/package.d
Comment (Geod24) [inline std/format/package.d:147]: [The `printf` manual uses the term "Conversion specifier"](https://man7.org/linux/man-pages/man3/printf.3.html):
> Each _conversion specification_ is introduced by the character %, and ends with a _conversion specifier_.

**Review by Geod24 — CHANGES_REQUESTED** [review]

File: std/format/package.d
Comment (berni44) [reply std/format/package.d:151]: Your sentence is a good idea, I'll add it. But I feel unwell with removing my suggestion - this sentence was meant to define the purpose of the format indicator, which IMHO would be missing if this sentence is removed.

File: std/format/package.d
Comment (berni44) [reply std/format/package.d:147]: When I started to rewrite the docs here, I encountered different namings for the same thing at various places. For example have a look at [printf in C++](https://www.cplusplus.com/reference/cstdio/printf/). It uses the word "specifier" for the whole thing as well as the letter at the end. In the former docs of `std.format` it wasn't used consistent either.

So I tried to unify the wording throughout the whole module, which resulted in a rewrite of the grammar (see PR #7927). IMHO this wording is consistent in itself and changing a single term would make other changes necessary. For example I'm using "format specifier" for the whole thing. If we use "conversion specifier" here, this needs to be changed too, to avoid confusion.

Another thing, that comes to my mind: Although `printf` was originally the model for this package, `std.format` has already diverged over the time (grouping is done with comma, not with a tick; there is no need to tell about `long` values in D, `%s` was added, and so on) and in my opinion we should not cling too strong to `printf` - it has some drawbacks (e.g. `%p` is unsafe and luckily was never added to Phobos). IMHO D can do better. :-)

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
@Geod24 OK to merge this?

**Comment by berni44** [comment]
@RazvanN7 Why is this not merged now? Can you have a look please?

**Comment by thewilsonator** [comment]
Because the request for changes from @Geod24 is still there

**Comment by Geod24** [comment]
@berni44 : IMO using yet another wording for this is worse than picking one out of two existing wordings.
I would favor C has it is more established and widespread, but I don't have a personal preference.

I however don't think that this concern should block the good work you're doing documenting this module, so I'll dismiss my review and remove `auto-merge` to give you time to read this reply, and decide whether or not you want to change it. If you don't, feel free to re-add `auto-merge` yourself. I've invited your back in the org, please don't delete your account again ;)

**Review by Geod24 — APPROVED** [review]

**Comment by berni44** [comment]
> @berni44 : IMO using yet another wording for this is worse than picking one out of two existing wordings.
> I would favor C has it is more established and widespread, but I don't have a personal preference.

I agree, that using what is there, is a good idea in general. However, I think, that the wording of C here is problematic, because some words, like "conversion specification" and "conversion specifier" can be easily confused (it actually took me a while to notice, that these two are two different things). And an other reason is, that I would need to reword all the places, where docs have changed already. I'm a little bit lazy here...

> I however don't think that this concern should block the good work you're doing documenting this module, so I'll dismiss my review and remove `auto-merge` to give you time to read this reply, and decide whether or not you want to change it. If you don't, feel free to re-add `auto-merge` yourself.

Unfortunately I've got no write access, so I can't. (Aside from not being able to add and remove certain buttons, I'm happy with having not write access, because that makes me feel safe that I cannot break anything accidentally.)

**Comment by Geod24** [comment]
> Unfortunately I've got no write access, so I can't. (Aside from not being able to add and remove certain buttons, I'm happy with having not write access, because that makes me feel safe that I cannot break anything accidentally.)

Check your emails. We have protection in place for master and stable so you can't screw those up. But if you really don't like the idea, just reject the invite.

**Pull request merged**
